src/printf/printf.c

```diff
@@ -831,89 +915,106 @@ static void print_decimal_number(output_gadget_t* output, floating_point_t numbe
+  /*
+   * The implementation follows David Gay (https://www.ampl.com/netlib/fp/dtoa.c).
+   *
+   * Since log_10 ( M * 2^x ) = log_10(M) + x , we can separate the components of
+   * our input number, and need only solve log_10(M) for M between 1 and 2 (as
+   * the base-2 mantissa is always 1-point-something). In that limited range, a
+   * Taylor series expansion of log10(x) should serve us well enough; and we'll
+   * take the mid-point, 1.5, as the point of expansion.
+   */
 
   floating_point_with_bit_access dwba = get_bit_access(positive_number);
-  // based on the algorithm by David Gay (https://www.ampl.com/netlib/fp/dtoa.c)
+  /* based on the algorithm by David Gay (https://www.ampl.com/netlib/fp/dtoa.c) */
   int exp2 = get_exp2(dwba);
-  // drop the exponent, so dwba.F comes into the range [1,2)
+  floating_point_t z;
+  /* drop the exponent, so dwba.F comes into the range [1,2) */
   dwba.U = (dwba.U & (((printf_fp_uint_t) (1) << FP_TYPE_STORED_MANTISSA_BITS) - 1U)) |
            ((printf_fp_uint_t) FP_TYPE_BASE_EXPONENT << FP_TYPE_STORED_MANTISSA_BITS);
-  floating_point_t z = (dwba.F - (floating_point_t) 1.5);
+  z = (dwba.F - (floating_point_t) 1.5);
   return (
-    // Taylor expansion around 1.5:
-              (floating_point_t) 0.1760912590556812420           // Expansion term 0: ln(1.5)            / ln(10)
-    + z     * (floating_point_t) 0.2895296546021678851 // Expansion term 1: (M - 1.5)   * 2/3  / ln(10)
+    /* Taylor expansion around 1.5: */
+              (floating_point_t) 0.1760912590556812420 /* Expansion term 0: ln(1.5)            / ln(10) */
+    + z     * (floating_point_t) 0.2895296546021678851 /* Expansion term 1: (M - 1.5)   * 2/3  / ln(10) */
 #if PRINTF_LOG10_TAYLOR_TERMS > 2
-    - z*z   * (floating_point_t) 0.0965098848673892950 // Expansion term 2: (M - 1.5)^2 * 2/9  / ln(10)
+    - z*z   * (floating_point_t) 0.0965098848673892950 /* Expansion term 2: (M - 1.5)^2 * 2/9  / ln(10) */
 #if PRINTF_LOG10_TAYLOR_TERMS > 3
-    + z*z*z * (floating_point_t) 0.0428932821632841311 // Expansion term 2: (M - 1.5)^3 * 8/81 / ln(10)
+    + z*z*z * (floating_point_t) 0.0428932821632841311 /* Expansion term 2: (M - 1.5)^3 * 8/81 / ln(10) */
 #endif
 #endif
-    // exact log_2 of the exponent x, with logarithm base change
-    + (floating_point_t) exp2 * (floating_point_t) 0.30102999566398119521 // = exp2 * log_10(2) = exp2 * ln(2)/ln(10)
+    /* exact log_2 of the exponent x, with logarithm base change */
+    + (floating_point_t) exp2 * (floating_point_t) 0.30102999566398119521 /* = exp2 * log_10(2) = exp2 * ln(2)/ln(10) */
   );
 }
 
 
 static floating_point_t pow10_of_int(int floored_exp10)
 {
-  // A crude hack for avoiding undesired behavior with barely-normal or slightly-subnormal values.
+  /* A crude hack for avoiding undesired behavior with barely-normal or slightly-subnormal values. */
   if (floored_exp10 == FP_TYPE_MAX_SUBNORMAL_EXPONENT_OF_10) {
     return FP_TYPE_MAX_SUBNORMAL_POWER_OF_10;
+  } else {
```

Comment:
here and elsewhere - drop gratuituous else keywords.

In fact, please move variable definitions to the top of the relevant block, so that we don't need the gratuious indents.